Bindgen padding errors targeting i686-pc-windows-msvc

After adding the dependency to this crate, a normal release build targeting x86_64 results in:

```rust
error[E0063]: missing fields `__bindgen_padding_0`, `__bindgen_padding_1` in initializer of `discord_game_sdk_sys::DiscordImageHandle`
  --> C:\Users\romet\.cargo\registry\src\github.com-1ecc6299db9ec823\discord_game_sdk-1.0.0-rc.1\src\user.rs:38:21
   |
38 |         ImageHandle(sys::DiscordImageHandle {
   |                     ^^^^^^^^^^^^^^^^^^^^^^^ missing `__bindgen_padding_0`, `__bindgen_padding_1`
```

Could be fixed by the next release of bindgen, as some padding issues were dealt with in an unreleased commit.

Bindgen next release tracker issue(?):
https://github.com/rust-lang/rust-bindgen/issues/1701

(merged) bindgen pull request potentially fixing this issue:
https://github.com/rust-lang/rust-bindgen/pull/1710

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -76,14 +76,15 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --all --examples --tests --target=${{ matrix.target }}
+          args: --all --all-targets --target=${{ matrix.target }}
         env:
           DISCORD_GAME_SDK_PATH: ${{ runner.temp }}
 
       - name: Test
         uses: actions-rs/cargo@v1
         with:
-          command: test --target=${{ matrix.target }}
+          command: test
+          args: --target=${{ matrix.target }}
         env:
           DISCORD_GAME_SDK_PATH: ${{ runner.temp }}
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -10,22 +10,27 @@ jobs:
       fail-fast: false
       matrix:
         target:
+          - x86_64-unknown-linux-gnu
+          - x86_64-apple-darwin
           - i686-pc-windows-gnu
           - i686-pc-windows-msvc
-          - x86_64-apple-darwin
           - x86_64-pc-windows-gnu
           - x86_64-pc-windows-msvc
-          - x86_64-unknown-linux-gnu
         rust:
           - nightly
           - stable
           - 1.37.0
 
-    runs-on: >
-      ${{ (contains(matrix.target, 'linux') && 'ubuntu-latest')
-       || (contains(matrix.target, 'apple') && 'macos-latest')
-       || 'windows-latest'
-       }}
+    env:
+      target-to-os:
+        x86_64-unknown-linux-gnu: 'ubuntu-latest'
+        x86_64-apple-darwin: 'macos-latest'
+        i686-pc-windows-gnu: 'windows-latest'
+        i686-pc-windows-msvc: 'windows-latest'
+        x86_64-pc-windows-gnu: 'windows-latest'
+        x86_64-pc-windows-msvc: 'windows-latest'
+
+    runs-on: ${{ env.target-to-os[matrix.target] }}
 
     steps:
       - name: Checkout

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,7 +21,7 @@ jobs:
           - stable
           - 1.37.0
 
-    runs-on: ${{ (contains(matrix.target, "linux") && "ubuntu-latest") || (contains(matrix.target, "apple") && "macos-latest") || "windows-latest" }}
+    runs-on: ${{ (contains(matrix.target, 'linux') && 'ubuntu-latest') || (contains(matrix.target, 'apple') && 'macos-latest') || 'windows-latest' }}
 
     steps:
       - name: Checkout

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,7 +21,11 @@ jobs:
           - stable
           - 1.37.0
 
-    runs-on: ${{ (contains(matrix.target, 'linux') && 'ubuntu-latest') || (contains(matrix.target, 'apple') && 'macos-latest') || 'windows-latest' }}
+    runs-on: >
+      ${{ (contains(matrix.target, 'linux') && 'ubuntu-latest')
+       || (contains(matrix.target, 'apple') && 'macos-latest')
+       || 'windows-latest'
+       }}
 
     steps:
       - name: Checkout
@@ -67,6 +71,7 @@ jobs:
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
+          target: ${{ matrix.target }}
           toolchain: ${{ matrix.rust }}
           override: true
           components: rustfmt, clippy
@@ -75,14 +80,14 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --all --examples --tests
+          args: --all --examples --tests --target=${{ matrix.target }}
         env:
           DISCORD_GAME_SDK_PATH: ${{ runner.temp }}
 
       - name: Test
         uses: actions-rs/cargo@v1
         with:
-          command: test
+          command: test --target=${{ matrix.target }}
         env:
           DISCORD_GAME_SDK_PATH: ${{ runner.temp }}
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -12,10 +12,10 @@ jobs:
         target:
           - x86_64-unknown-linux-gnu
           - x86_64-apple-darwin
-          - i686-pc-windows-gnu
           - i686-pc-windows-msvc
-          - x86_64-pc-windows-gnu
           - x86_64-pc-windows-msvc
+        # - i686-pc-windows-gnu #65
+        # - x86_64-pc-windows-gnu #65
         rust:
           - nightly
           - stable

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -9,16 +9,19 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-latest
-          - macos-latest
-          - windows-latest
+        target:
+          - i686-pc-windows-gnu
+          - i686-pc-windows-msvc
+          - x86_64-apple-darwin
+          - x86_64-pc-windows-gnu
+          - x86_64-pc-windows-msvc
+          - x86_64-unknown-linux-gnu
         rust:
-          - stable
           - nightly
+          - stable
           - 1.37.0
 
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ (contains(matrix.target, "linux") && "ubuntu-latest") || (contains(matrix.target, "apple") && "macos-latest") || "windows-latest" }}
 
     steps:
       - name: Checkout
@@ -36,47 +39,29 @@ jobs:
 
       - name: Set DYLD_LIBRARY_PATH
         working-directory: ${{ runner.temp }}
-        if: startsWith(matrix.os, 'macos')
+        if: startsWith(runner.os, 'macos')
         run: |
           echo "::set-env name=DYLD_LIBRARY_PATH::$DYLD_LIBRARY_PATH:$PWD/lib/x86_64"
 
       - name: Cache libclang
         id: cache-libclang
-        if: startsWith(matrix.os, 'windows')
+        if: startsWith(runner.os, 'windows')
         uses: actions/cache@v1
         with:
           path: 'C:\just-libclang'
           key: windows-libclang
 
       - name: Install LLVM
-        if: startsWith(matrix.os, 'windows') && steps.cache-libclang.outputs.cache-hit != 'true'
+        if: startsWith(runner.os, 'windows') && steps.cache-libclang.outputs.cache-hit != 'true'
         run: |
           choco install llvm -r -y
           mkdir 'C:\just-libclang'
           cp 'C:\Program Files\LLVM\bin\libclang.dll' 'C:\just-libclang'
 
       - name: Set LIBCLANG_PATH
-        if: startsWith(matrix.os, 'windows')
+        if: startsWith(runner.os, 'windows')
         run: |
           echo '::set-env name=LIBCLANG_PATH::C:\just-libclang'
-
-      - name: Cache cargo registry
-        uses: actions/cache@v1
-        with:
-          path: ~/.cargo/registry
-          key: ${{ runner.os }}-${{ matrix.rust }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
-
-      - name: Cache cargo index
-        uses: actions/cache@v1
-        with:
-          path: ~/.cargo/git
-          key: ${{ runner.os }}-${{ matrix.rust }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
-
-      - name: Cache cargo build
-        uses: actions/cache@v1
-        with:
-          path: target
-          key: ${{ runner.os }}-${{ matrix.rust }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
 
       - name: Set up Rust
         uses: actions-rs/toolchain@v1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,16 +21,7 @@ jobs:
           - stable
           - 1.37.0
 
-    env:
-      target-to-os:
-        x86_64-unknown-linux-gnu: 'ubuntu-latest'
-        x86_64-apple-darwin: 'macos-latest'
-        i686-pc-windows-gnu: 'windows-latest'
-        i686-pc-windows-msvc: 'windows-latest'
-        x86_64-pc-windows-gnu: 'windows-latest'
-        x86_64-pc-windows-msvc: 'windows-latest'
-
-    runs-on: ${{ env.target-to-os[matrix.target] }}
+    runs-on: ${{ (contains(matrix.target, 'linux') && 'ubuntu-latest') || (contains(matrix.target, 'apple') && 'macos-latest') || 'windows-latest'}}
 
     steps:
       - name: Checkout

--- a/src/image_handle.rs
+++ b/src/image_handle.rs
@@ -29,6 +29,7 @@ impl ImageHandle {
             type_: ImageKind::User.into(),
             id: user_id,
             size,
+            ..Default::default()
         })
     }
 }

--- a/src/image_handle.rs
+++ b/src/image_handle.rs
@@ -25,12 +25,13 @@ impl ImageHandle {
 
     /// Create new Image Handle
     pub fn from_user_id(user_id: UserID, size: u32) -> Self {
-        Self(sys::DiscordImageHandle {
-            type_: ImageKind::User.into(),
-            id: user_id,
-            size,
-            ..Default::default()
-        })
+        let mut handle = sys::DiscordImageHandle::default();
+
+        handle.type_ = ImageKind::User.into();
+        handle.id = user_id;
+        handle.size = size;
+
+        Self(handle)
     }
 }
 

--- a/src/user.rs
+++ b/src/user.rs
@@ -35,11 +35,7 @@ impl User {
 
     /// Create an [Image Handle](struct.ImageHandle.html) targeting the user's avatar
     pub fn image_handle(&self, size: u32) -> ImageHandle {
-        ImageHandle(sys::DiscordImageHandle {
-            type_: sys::DiscordImageType_User,
-            id: self.0.id,
-            size,
-        })
+        ImageHandle::from_user_id(self.id(), size)
     }
 }
 

